chia/plotting/manager.py

```diff
@@ -95,12 +149,19 @@ def plot_count(self):
         with self:
             return len(self.plots)
 
+    def refresh_interval_exceeded(self, last_refresh_time: float) -> bool:
+        return time.time() - last_refresh_time > float(self.refresh_parameter.interval_seconds)
+
     def needs_refresh(self) -> bool:
-        return time.time() - self.last_refresh_time > float(self.refresh_parameter.interval_seconds)
+        return self.refresh_interval_exceeded(self.last_refresh_time)
+
+    def needs_save_cache(self) -> bool:
+        return self.refresh_interval_exceeded(self.last_save_cache_time)
```

Comment:
this looks like it's mixing up the `last_save_cache_time` and `last_refresh_time`. It's not obvious that it's correct to compare/interchange those two timestamps.

But, given that we already have a *different* mechanism for refreshing the plots periodically, it seems it would make more sense to save the cache every time we add more plots. There's no point in saving the cache if we didn't pick up any new plots.

Reply:
> this looks like it's mixing up the `last_save_cache_time` and `last_refresh_time`. It's not obvious that it's correct to compare/interchange those two timestamps.

How does it look that they are mixed up? They just both get compared to the refresh interval, isn't that pretty obvious by the name of all related methods and variables?

> There's no point in saving the cache if we didn't pick up any new plots.

Where do you see this?

Reply:
> How does it look that they are mixed up?
in the function on line 152, `refresh_interval_exceeds()`, the parameter name is `last_refresh_time`. You're passing something in that isn't that. That's what I mean.

>> There's no point in saving the cache if we didn't pick up any new plots.
> Where do you see this?
I don't think I understand what you're asking. I'm saying that I don't see a reason to save the cache on a timer at all. So as far as I can tell, it would be an improvement to remove `last_save_cache_time` and `needs_save_cache()` and all their uses.

Reply:
https://github.com/Chia-Network/chia-blockchain/pull/8009#issuecomment-909086002